### PR TITLE
Bug 1316195: obscure secret payloads

### PR DESCRIPTION
**Pull request by djmitche**

### CODE REVIEW

**Comment by djmitche** [comment]
Errors are because it's not installing libxml.  *hate*

The tests run successfully for me.

File: lib/api.js [hunk 1]
```diff
@@ -32,6 +32,10 @@ api.declare({
   scopes:      [['secrets:set:<name>']],
   title:       'Set Secret',
   stability:    'stable',
+  cleanPayload: payload => {
```
Comment (jonasfj) [inline lib/api.js]: Isn't it safer to do: `cleanPayload: payload => _.defaults({secret: '(omitted)'}, payload),`

Comment (djmitche) [inline lib/api.js]: Not necessarily, since some payloads don't include a secret -- that might confuse the user!

Comment (djmitche) [inline lib/api.js]: Oh, that's silly.  Your method is clearer, but I think there's a method involving "..." that's even better.. let me play.

Comment (djmitche) [inline lib/api.js]: Hah, well, that would work if we were using ES6.  This service hasn't seen much love!

File: lib/api.js [hunk 2]
```diff
@@ -32,6 +32,7 @@ api.declare({
   scopes:      [['secrets:set:<name>']],
   title:       'Set Secret',
   stability:    'stable',
+  cleanPayload: payload => ({...payload, secret: "(OMITTED)"}),
```
Comment (djmitche) [inline lib/api.js]: There we go!

**Pull request merged**